rpc/core/env.go

```diff
@@ -245,3 +261,109 @@ func (env *Environment) latestUncommittedHeight() int64 {
 	}
 	return env.BlockStore.Height() + 1
 }
+
+// fileSize returns the size of the file at the given path.
+func fileSize(fPath string) (int, error) {
+	// we use os.Stat here instead of os.ReadFile, because we don't want to load
+	// the entire file into memory just to compute its size from the resulting
+	// []byte slice.
+	fInfo, err := os.Stat(fPath)
+	if errors.Is(err, fs.ErrNotExist) {
+		return 0, fmt.Errorf("the file is unavailable at %s", fPath)
+	} else if err != nil {
+		return 0, fmt.Errorf("accessing file: %s", err)
+	}
+	return int(fInfo.Size()), nil
+}
+
+// mkChunksDir creates a new directory to store the genesis file's chunks.
+// gFilePath is the genesis file's full path on disk.
+// dirName is the name of the directory to be created, not it's path on disk.
+// mkChunksDir will create a directory named 'dirName' as a sub-directory of the
+// genesis file's directory (gFileDir).
+// It returns the new directory's full path or an empty string if there is an
+// error.
+func mkChunksDir(gFilePath string, dirName string) (string, error) {
+	var (
+		gFileDir = filepath.Dir(gFilePath)
+		dirPath  = filepath.Join(gFileDir, dirName)
+	)
+	if _, err := os.Stat(dirPath); err == nil {
+		// directory already exists; this might happen it the node crashed and
+		// could not do cleanup. Delete it to start from scratch.
+		if err := os.RemoveAll(dirPath); err != nil {
+			return "", fmt.Errorf("deleting existing chunks directory: %s", err)
+		}
+	} else if !os.IsNotExist(err) {
+		return "", fmt.Errorf("accessing directory: %s", err)
+	}
+
+	if err := os.Mkdir(dirPath, 0o755); err != nil {
```

Comment:
why 755?

Reply:
755 should be read, write, and execute permission to the owner, but only read and execute for others. 
The owner does need all three because we create and write the files to store in the directory at runtime, and later we want to read them.

Do you think it should be 700? i.e., no permissions for others?

Reply:
I think it's fine, but yeah probably others don't need to see that.

Reply:
We can make the change 👍